migrations/20171024163252-host-cleanup.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+const Promise = require('bluebird');
+
+const DRY_RUN = true;
+
+const insert = (sequelize, table, entry) => {
+  delete entry.id;
+  console.log(`INSERT INTO "${table}" ("${Object.keys(entry).join('","')}") VALUES (:${Object.keys(entry).join(",:")})`)
+  if (entry.data) {
+    entry.data = JSON.stringify(entry.data);
+  }
+  return sequelize.query(`
+    INSERT INTO "${table}" ("${Object.keys(entry).join('","')}") VALUES (:${Object.keys(entry).join(",:")})
+  `, { replacements: entry });
+}
+
+/*
+1. Find all supercollectives that are listed as hosts themselves
```

Comment:
Why only super collectives and not all hosts?

Reply:
All the confusing places are where a host is supercollective. I initially ran it for all Hosts and had same result.

Reply:
Scratch that - rest of the hosts are fine. Some of their transactions are not though - wrote a health check script that we can run periodically to point out inconsistencies. So far,

PREMIGRATION STATS

>>> Checking Hosts to be USER or ORG...
	>>> Hosts found:  100
	>>> Hosts found with incorrect type:  10

>>> Checking Host Collectives...
	>>> Self-referencing Hosts found:  9

>>> Checking USER and ORG Collectives
	>>> USER or ORGs found with HostCollectiveId:  0

>>> Checking Transactions...
	>>> Transactions without `FromCollectiveId`: 550
	>>> Transactions with same source and destination:  82
	>>> Transactions without `TransactionGroup`:  1456
	>>> Orders with odd (not multiples of 2) number of transactions:  260
	>>> Expenses with less than or more than 2 transactions:  2
	>>> Transaction groups that are not pairs:  524
	>>> Transactions without OrderId or ExpenseId:  898


—— POSTMIGRATION STATS —— 

>>> Checking Hosts to be USER or ORG...
	>>> Hosts found:  100
	>>> Hosts found with incorrect type:  1

>>> Checking Host Collectives...
	>>> Self-referencing Hosts found:  0

>>> Checking USER and ORG Collectives
	>>> USER or ORGs found with HostCollectiveId:  0

>>> Checking Transactions...
	>>> Transactions without `FromCollectiveId`: 110
	>>> Transactions with same source and destination:  34
	>>> Transactions without `TransactionGroup`:  1458 (this is a bug from v2 release, fixed)
	>>> Orders with odd (% 2 check) number of transactions:  59
	>>> Expenses with less than or more than 2 transactions:  2
	>>> Transaction groups that are not pairs:  84
	>>> Transactions without OrderId or ExpenseId 659

I'm going to try to fix as many of these via Migration. Rest we'll deal with manually.

Reply:
We are close, rest of these are one-offs that can be fixed manually:

>>> Checking Hosts to be USER or ORG...
	>>> Hosts found:  100
	>>> Hosts found with incorrect type:  0

>>> Checking Host Collectives...
	>>> Self-referencing Hosts found:  0

>>> Checking USER and ORG Collectives
	>>> USER or ORGs found with HostCollectiveId:  0

>>> Checking Transactions...
	>>> Transactions without `FromCollectiveId`: 1
	>>> Transactions with same source and destination:  34
	>>> Transactions without `TransactionGroup`:  0
	>>> Orders with odd (not multiple of 2) number of transactions:  0
	>>> Expenses with less than or more than 2 transactions:  2
	>>> Transaction groups that are not pairs:  0
	>>> Transactions without OrderId or ExpenseId:  608